main.go

```diff
@@ -220,6 +220,7 @@ func main() {
 		MaxConcurrentReconciles:   concurrent,
 		DependencyRequeueInterval: requeueDependency,
 		RateLimiter:               helper.GetRateLimiter(rateLimiterOptions),
+		RecoverPanic:              true,
```

Comment:
Instead of using the literal value, please create a `const` at the top of the file.

Reply:
I'd also consider if we need to configure this here or just set it directly in the `controller.Options` as we don't have any reason to change this value.

Reply:
Should I go ahead with @pjbgf's suggestion and [implementation here](https://github.com/fluxcd/image-automation-controller/pull/416#issue-1334304830) or wait for some guidance on @darkowlzz's point?

Reply:
@Santosh1176 let's go ahead with @darkowlzz suggestion, as that simplifies the implementation.